ark/mibi/tiling_utils.py

```diff
@@ -82,69 +93,195 @@ def generate_region_info(region_start_x, region_start_y, fov_num_x, fov_num_y,
-    """Given a file specifying fov regions, set the MIBI tiling parameters
+def read_tma_region_input(fov_tile_info, region_start_x, region_start_y,
+                          region_end_x, region_end_y, fov_num_x, fov_num_y, x_fov_size,
+                          y_fov_size, x_intervals, y_intervals, region_rand):
+    """Reads input for TMAs from user and fov_tile_info
 
-    User inputs will be required for many values. Also returns moly_path data.
+    Updates all the tiling params inplace
 
     Args:
-        fov_list_path (str):
-            Path to the JSON file containing the fovs used to define each tiled region
-        moly_path (str):
-            Path to the JSON moly point file, needed to separate fovs
-
-    Returns:
-        tuple:
-            Contains:
-
-            - A dict containing the tiling parameters for each fov
-            - A dict defining the moly points to insert if specified
+        fov_tile_info (dict):
+            The data containing the fovs used to define each tiled region
+        region_start_x (list):
+            List of x starting points sorted by run
+        region_start_y (list):
+            List of y starting points sorted by run
+        region_end_x (list):
+            List of x ending points sorted by run
+        region_end_y (list):
+            List of y ending points sorted by run
+        fov_num_x (list):
+            Number of fovs along the x coord sorted by run
+        fov_num_y (list):
+            Number of fovs along the y coord sorted by run
+        x_fov_size (list):
+            The size of the x axis sorted by run
+        y_fov_size (list):
+            The size of the y axis sorted by run
+        x_intervals (list):
+            The indices to put fovs at sorted by run
+        y_intervals (list):
+            The indices to put fovs at sorted by run
+        region_rand (list):
+            Whether to set randomization or not sorted by run
     """
 
-    # file path validation
-    if not os.path.exists(fov_list_path):
-        raise FileNotFoundError("FOV region file %s does not exist" % fov_list_path)
+    # there has to be a starting and ending fov for each region
+    if len(fov_tile_info['fovs']) % 2 != 0:
+        raise ValueError(
+            "Data in fov_list_path needs to contain a start and end fov for each region"
+        )
 
-    if not os.path.exists(moly_path):
-        raise FileNotFoundError("Moly point file %s does not exist" % moly_path)
+    # every two fovs should define the start and end of the fov
+    for i in range(0, len(fov_tile_info['fovs']), 2):
+        # define the current start and end fov
+        fov_batches = fov_tile_info['fovs'][i:i + 2]
+        start_fov = fov_batches[0]
+        end_fov = fov_batches[1]
+
+        # the names of the fovs need to match
+        if start_fov['name'] != end_fov['name']:
+            err_msg = "FOV mismatch at indices %d, %d: names %s, %s are not equal"
```

Comment:
Is this necessary for the script to work? If not, I don't think it's needed as input validation, we don't care what the points are called.

Reply:
@ngreenwald true, we don't really need this error check.